### DISCUSSION TITLE
Generating a TLS certificate on MicroPython

### DISCUSSION THREAD

**Discussion by jouellnyc** [micropython/micropython#9687] [Libraries & Drivers] [answered]
Hi Everyone

My research is suggesting that there is no method to create a TLS cert using current micropython libraries on the  micro-controller itself. I'm trying to use it for the  [microdot](https://microdot.readthedocs.io/en/latest/api.html#module-microdot_ssl) server I have running .  Is that correct?

My use case is that I would like to send a micro-controller to my father/family and have him be able to set it up through a short "wizard" workflow that I have set up instead him using USB/TTL/etc. I.E Avoiding manual steps. One of the steps is submitting the Wi-Fi password of his network at setup time so that the micro-controller will then be used to connect to his network. 

So automating an tls cert generation would be a nice way to finalize this. 

(I could just  send him one pre configured, but where's the fun in that :) )

As always thanks in advance!

**Comment by jimmo** [comment, marked as answer]
That's right, there isn't currently a way to generate certs on-device. Would be neat but I'm not sure the code size increase would be justified.

**Reply by jouellnyc** [reply]
Thanks jimmo

Right. I'm not sure it is either. But that again, maybe I am the first of thousands looking to do this :) At least for this use case there are workarounds and mitigations as the AP still uses WPA2-PSK.  And if  someone really wanted, they could go into a room with little connectivity like a basement and submit the wifi cred for some added protection. 

Again, thanks for the reply.
